includes/admin/class-wc-subscriptions-admin.php

```diff
@@ -509,11 +509,11 @@ public static function save_subscription_meta( $post_id ) {
 		update_post_meta( $post_id, '_regular_price', $subscription_price );
 		update_post_meta( $post_id, '_sale_price', $sale_price );
 
-		$site_offset = get_option( 'gmt_offset' ) * 3600;
+		$site_offset = wc_timezone_offset();
 
 		// Save the timestamps in UTC time, the way WC does it.
-		$date_from = ( ! empty( $_POST['_sale_price_dates_from'] ) ) ? wcs_date_to_time( $_POST['_sale_price_dates_from'] ) - $site_offset : '';
-		$date_to   = ( ! empty( $_POST['_sale_price_dates_to'] ) ) ? wcs_date_to_time( $_POST['_sale_price_dates_to'] ) - $site_offset : '';
+		$date_from = ! empty( $_POST['_sale_price_dates_from'] ) ? wcs_date_to_time( gmdate( 'Y-m-d 00:00:00', wcs_strtotime_dark_knight( wc_clean( wp_unslash( $_POST['_sale_price_dates_from'] ) ) ) ) ) - $site_offset : '';
```

Comment:
Instead of using `gmdate` and `wcs_strtotime_dark_knight()` functions which force UTC, we could simply use `date()` and `strtotime` since `wcs_date_to_time()` will also convert the time to UTC, but that then throws PHPCS issues and so we'd need to add `// phpcs:ignore WordPress.DateTime.RestrictedFunctions.date_date`

I opted to just use gmdate and our dark_knight functions so that we don't second guess later on whether there's an issue with these lines of code not using UTC functions.

Any thoughts on this ?

Reply:
I was going to suggest using `wc_string_to_timestamp()` as that's a WC core function that serves the same purpose. However, I was still concerned because we'd be handling this differently to WooCommerce core and we be at a risk of having this issue return if WC changed the way they calculated these dates. 

It made me think though that maybe we don't even need to set these dates and we can just rely on WC doing it for us. 

I don't know when this changed, but that does look to be the case. We can just delete the two lines below.

```php
update_post_meta( $post_id, '_sale_price_dates_from', $date_from );
update_post_meta( $post_id, '_sale_price_dates_to', $date_to );
```